File: lint-arwaky/src-python/surfaces/cli_setup_controller.py
"""Helper functions for CLI setup commands."""

from ..taxonomy import DirectoryPath, EnvContentVO, McpConfigVO
from ..contract import ServiceContainerAggregate


class SetupManagementSurface:
    """Surface for setup management logic."""

    @property
    def _INTERFACE(self) -> object:
        """ARCHITECTURAL COMMITMENT: Required interface."""
        return ServiceContainerAggregate

    container: ServiceContainerAggregate | None = None

    def __init__(self):
        """Initialize surface."""
        pass

    def register_all(self, container: ServiceContainerAggregate) -> None:
        """Setup management registration fulfillment."""
        self.container = container

    def generate_env(self, home: DirectoryPath) -> EnvContentVO:
        """Generate .env content via container."""
        if not self.container:
            raise RuntimeError("SetupManagementSurface container not initialized")

        content = self.container.setup_processor.generate_env(home)
        return content

    def generate_mcp_config(self) -> McpConfigVO:
        """Generate mcp.json entry via container."""
        if not self.container:
            raise RuntimeError("SetupManagementSurface container not initialized")

        config = self.container.setup_processor.generate_mcp_config()
        return config

    def mcp_config_claude(self) -> McpConfigVO:
        """MCP config for Claude Desktop."""
        return self.generate_mcp_config()

    def mcp_config_hermes(self) -> McpConfigVO:
        """MCP config for Hermes Agent."""
        return self.generate_mcp_config()

    def mcp_config_vscode(self) -> McpConfigVO:
        """MCP config for VS Code."""
        base = self.generate_mcp_config().value.value
        return McpConfigVO(value={"mcp": {"servers": base}})


def register_setup_management(container: ServiceContainerAggregate) -> None:
    """Register setup management surface."""
    _get_instance().register_all(container)


# Lazy singleton — created on first call to avoid import-time side effects
_Instance = None


def _get_instance():
    global _Instance
    if _Instance is None:
        _Instance = SetupManagementSurface()
    return _Instance


def generate_env(*args, **kwargs):
    return _get_instance().generate_env(*args, **kwargs)


def generate_mcp_config(*args, **kwargs):
    return _get_instance().generate_mcp_config(*args, **kwargs)


def mcp_config_claude(*args, **kwargs):
    return _get_instance().mcp_config_claude(*args, **kwargs)


def mcp_config_hermes(*args, **kwargs):
    return _get_instance().mcp_config_hermes(*args, **kwargs)


def mcp_config_vscode(*args, **kwargs):
    return _get_instance().mcp_config_vscode(*args, **kwargs)

File: lint-arwaky/src-rust/surfaces/cli_setup_controller.rs
/// Helper functions for CLI setup commands.

use crate::taxonomy::*;
use crate::contract::*;

pub struct SetupManagementSurface {
    pub container: Option<ServiceContainerAggregate>,
}

impl SetupManagementSurface {
    pub fn new() -> Self {
        Self { container: None }
    }

    pub fn register_all(&mut self, container: ServiceContainerAggregate) {
        self.container = Some(container);
    }

    pub fn generate_env(&self, home: &str) -> String {
        // In real impl: container.setup_processor.generate_env(home)
        format!(
            "# Auto-Linter environment configuration\nHOME={home}\nMCP_LOG_LEVEL=INFO\n"
        )
    }

    pub fn generate_mcp_config(&self) -> String {
        // In real impl: container.setup_processor.generate_mcp_config()
        r#"{
  "mcpServers": {
    "auto-linter": {
      "command": "auto-linter",
      "args": []
    }
  }
}"#.to_string()
    }

    pub fn mcp_config_claude(&self) -> String {
        self.generate_mcp_config()
    }

    pub fn mcp_config_hermes(&self) -> String {
        self.generate_mcp_config()
    }

    pub fn mcp_config_vscode(&self) -> String {
        let base = self.generate_mcp_config();
        format!("{{\"mcp\": {{\"servers\": {base}}}}}")
    }
}

// Lazy singleton
static INSTANCE: std::sync::Mutex<Option<SetupManagementSurface>> = std::sync::Mutex::new(None);

fn get_instance() -> std::sync::MutexGuard<'static, Option<SetupManagementSurface>> {
    let mut guard = INSTANCE.lock().unwrap();
    if guard.is_none() {
        *guard = Some(SetupManagementSurface::new());
    }
    guard
}

pub fn register_setup_management(container: ServiceContainerAggregate) {
    let mut guard = INSTANCE.lock().unwrap();
    if let Some(ref mut s) = *guard {
        s.register_all(container);
    } else {
        let mut s = SetupManagementSurface::new();
        s.register_all(container);
        *guard = Some(s);
    }
}

pub fn generate_env(home: &str) -> String {
    let guard = get_instance();
    guard.as_ref().map(|s| s.generate_env(home)).unwrap_or_default()
}

pub fn generate_mcp_config() -> String {
    let guard = get_instance();
    guard.as_ref().map(|s| s.generate_mcp_config()).unwrap_or_default()
}

pub fn mcp_config_claude() -> String {
    let guard = get_instance();
    guard.as_ref().map(|s| s.mcp_config_claude()).unwrap_or_default()
}

pub fn mcp_config_hermes() -> String {
    let guard = get_instance();
    guard.as_ref().map(|s| s.mcp_config_hermes()).unwrap_or_default()
}

pub fn mcp_config_vscode() -> String {
    let guard = get_instance();
    guard.as_ref().map(|s| s.mcp_config_vscode()).unwrap_or_default()
}
